### PR TITLE
[video] CVideoInfoTag: Remove CVideoDataBase dependency. 

### PR DESCRIPTION
**Pull request by ksooo**
Improves per…formance of ListItem.IsVideoExtras GUI label implementation drastically, because no more db lookup needed.

Screenshot 2023-12-16 at 23 48 19

Runtime-tested on macOS, latest Kodi master.

@enen92 whenever you find some time for a review.

### CODE REVIEW

File: xbmc/video/VideoInfoTag.cpp [hunk 1]
```diff
@@ -738,6 +747,8 @@ void CVideoInfoTag::Serialize(CVariant& value) const
   value["tag"] = m_tags;
   value["videoversion"] = m_typeVideoVersion;
   value["videoversionid"] = m_idVideoVersion;
+  value["hasvideoversions"] = m_hasVideoVersions;
```
Comment (enen92) [inline xbmc/video/VideoInfoTag.cpp:750]: Needs minor bump on Json API as far as I remember

Comment (ksooo) [reply xbmc/video/VideoInfoTag.cpp:750]: Not sure, something else needs to be done to make the info available via the API, as far as I remember. Putting it here is just a prerequisite for the API, without further consequences. 

Maybe @DaveTBlake  can shed some light on this.

Intention of this PR is not to make the new stuff available via the API.

File: xbmc/video/VideoVersionTypes.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+enum class VideoVersionTypeOwner
```
[the remark this answers is not in the document]
Comment (ksooo) [reply xbmc/video/VideoVersionTypes.h:11]: That would be up to @xodidox. I just moved what we had to a separate header, without knowing the exact semantics.

File: xbmc/video/VideoInfoTag.cpp (hunk 1 above)
Comment (the-black-eagle) [reply xbmc/video/VideoInfoTag.cpp:750]: ^^ That.  There is much more to do to make it available via the API then just serialising it.  This PR doesn't touch the API so no need to bump the version.

Comment (enen92) [reply xbmc/video/VideoInfoTag.cpp:750]: Ok

**Review by enen92 — APPROVED** [review]

File: xbmc/video/VideoVersionTypes.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (xodidox) [reply xbmc/video/VideoVersionTypes.h:11]: I'll add doxygen comments for the enum.

**Pull request merged**